### PR TITLE
Fix reset passcode

### PR DESCRIPTION
**Pull request by MouazAlzahabi**
Handles #2912 

Changes proposed in this pull request:
- Fix reset passcode logic
- Refactor enter passcode screen handlers

### CODE REVIEW

File: Multisig/UI/Safe Management/Create Safe/Setup Safe/CreateSafeViewController.swift [hunk 1]
```diff
@@ -818,9 +818,11 @@ class CreateSafeViewController: UIViewController, UITableViewDelegate, UITableVi
                     } else {
                         self.sign()
                     }
-                } else if reset {
-                    self.dismiss(animated: false, completion: nil)
                 }
+                // TODO: Check if we need this
```
Comment (elgatovital) [inline Multisig/UI/Safe Management/Create Safe/Setup Safe/CreateSafeViewController.swift]: cant this be checked now?

File: Multisig/UI/Settings/Passcode/PasscodeProtecting.swift [hunk 2]
```diff
@@ -10,25 +10,30 @@ import Foundation
...
 }
 
 extension PasscodeProtecting where Self: UIViewController {
-    func authenticate(biometry: Bool = true, options: PasscodeOptions = [], completion: @escaping (_ success: Bool, _ reset: Bool) -> Void) {
+    func authenticate(biometry: Bool = true, options: PasscodeOptions = [], completion: @escaping (_ success: Bool) -> Void) {
         if App.shared.auth.isPasscodeSetAndAvailable && (options.isEmpty || !AppSettings.passcodeOptions.intersection(options).isEmpty) {
             let passcodeVC = EnterPasscodeViewController()
             passcodeVC.usesBiometry = biometry
-            passcodeVC.passcodeCompletion = { [weak self] success, reset, _ in
+            passcodeVC.passcodeCompletion = { [weak self] result in
```
Comment (elgatovital) [inline Multisig/UI/Settings/Passcode/PasscodeProtecting.swift:21]: how do we handle passcode reset now? (When user clicks on forgot passcode)

Comment (MouazAlzahabi) [reply Multisig/UI/Settings/Passcode/PasscodeProtecting.swift:21]: In EnterPasscodeViewController the trigger point, then we close the view as we usual, after I test there is no need to do more than that

File: Multisig/UI/Safe Management/Create Safe/Setup Safe/CreateSafeViewController.swift (hunk 1 above)
Comment (MouazAlzahabi) [reply Multisig/UI/Safe Management/Create Safe/Setup Safe/CreateSafeViewController.swift]: true will check

**Review by elgatovital — APPROVED** [review]

**Pull request merged**